Source/Core/Common/Concepts.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+// This feature detection be removed once Dolphin Emulator infrastructure finishes upgrading
+#ifndef __cpp_concepts
+#error "C++20 concepts support not detected!  Update your compiler!"
+#endif
+
+#include <limits>
+#include <type_traits>
+
+namespace Common
+{
+// These concepts are a terrible workaround for buildbots which support C++20 but have an outdated
+// C++ stdlib.  Remove these redundant concepts ASAP!
```

Comment:
Probably instead of saying "buildbots" say "compilers (including some of Dolphin's buildbots)". I think it would also be useful to mention what the equivalent standard one is (e.g. `std::same_as`, `std::floating_point`, etc).